core/src/main/java/io/questdb/cutlass/pgwire/modern/PGPipelineEntry.java

```diff
@@ -2330,14 +2317,22 @@ private void outRowDescription(PGResponseSink utf8Sink) {
                 utf8Sink.putNetworkShort((short) (i + 1)); //column number, starting from 1
                 utf8Sink.putNetworkInt(PGOids.getTypeOid(columnType)); // type
                 // type size
-                if (ColumnType.tagOf(columnType) < ColumnType.STRING) {
+                short typeTag = ColumnType.tagOf(columnType);
+                if (typeTag < ColumnType.STRING && typeTag != ColumnType.CHAR) {
```

Comment:
there is optimisation opportunity to not rely on `Column.sizeOf(columnType)` anymore. This was a bit of a hack initially. We diverged from our internal sizes before and now we diverge even more. Having a dedicated structure for PG type sizes can also eliminate `networkShort` call, that reverses bits.

Reply:
@bluestreak01 wdyt?

Reply:
very cool @jerrinot !